app/controllers/CheckYourAnswersController.scala

```diff
@@ -16,73 +16,144 @@
 
     val checkYourAnswersHelper = new CheckYourAnswersHelper(request.userAnswers, countriesService.getAllCountries, messagesApi, lang)
 
     val sections = Seq(
       AnswerSection(
-        Some("checkYourAnswers.applicantRegisteredSection"),
-        Seq(
-          checkYourAnswersHelper.registeredAddressForEori,
-          checkYourAnswersHelper.enterContactDetails
-//          checkYourAnswersHelper.whichBestDescribesYou
-        ).flatten
-      ),
-      AnswerSection(
-        Some("checkYourAnswers.applicantOtherBusiness"),
+        Some("checkYourAnswers.aboutTheGoodsSection"),
         Seq(
-          checkYourAnswersHelper.registerBusinessRepresenting,
-          checkYourAnswersHelper.uploadWrittenAuthorisation
-        ).flatten
-      ),
-      AnswerSection(
-        Some("checkYourAnswers.informationAboutYourItemSection"),
-        Seq(
-          checkYourAnswersHelper.selectApplicationType,
-          checkYourAnswersHelper.previousCommodityCode,
           checkYourAnswersHelper.provideGoodsName,
           checkYourAnswersHelper.provideGoodsDescription,
           checkYourAnswersHelper.addConfidentialInformation,
           checkYourAnswersHelper.provideConfidentialInformation,
           checkYourAnswersHelper.supportingMaterialFileListChoice,
           checkYourAnswersHelper.supportingMaterialFileList,
-          checkYourAnswersHelper.commodityCodeBestMatch,
-          checkYourAnswersHelper.commodityCodeDigits,
           checkYourAnswersHelper.whenToSendSample,
           checkYourAnswersHelper.isSampleHazardous,
           checkYourAnswersHelper.returnSamples,
-          checkYourAnswersHelper.similarItemCommodityCode,
-          checkYourAnswersHelper.commodityCodeRulingReference,
+          checkYourAnswersHelper.commodityCodeBestMatch,
+          checkYourAnswersHelper.commodityCodeDigits,
           checkYourAnswersHelper.legalChallenge,
           checkYourAnswersHelper.legalChallengeDetails
         ).flatten
+      ),
+      AnswerSection(
+        Some("checkYourAnswers.aboutOtherRulingsSection"),
+        Seq(
+          checkYourAnswersHelper.selectApplicationType,
+          checkYourAnswersHelper.previousCommodityCode,
+          checkYourAnswersHelper.similarItemCommodityCode,
+          checkYourAnswersHelper.commodityCodeRulingReference
+        ).flatten
+      ),
+      AnswerSection(
+        Some("checkYourAnswers.aboutTheApplicantSection"),
+        Seq(
+          checkYourAnswersHelper.enterContactDetailsName,
+          checkYourAnswersHelper.enterContactDetailsEmail,
+          checkYourAnswersHelper.enterContactDetailsPhone
+        ).flatten
       )
     )
 
     Ok(check_your_answers(appConfig, sections))
   }
 
+  def onSubmit(): Action[AnyContent] = (identify andThen getData).async { implicit request: OptionalDataRequest[_] =>
+
+    val answers = request.userAnswers.get // TODO: we should not call `get` on an Option
```

Comment:
If we add `andThen requireData` at the start of the action then we won't have to do the `.get` above

Reply:
Then the type of the `implicit request` above will be `DataRequest[_]` instead

Reply:
We've added those changes David

Reply:
Awesome! 👍 Looks good to go